src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -1537,6 +1541,76 @@ bool SystemDictionaryShared::is_dumptime_table_empty() {
   return false;
 }
 
+class CloneDumpTimeClassTable: public StackObj {
+  DumpTimeSharedClassTable* _table;
+  DumpTimeSharedClassTable* _cloned_table;
+ public:
+  CloneDumpTimeClassTable(DumpTimeSharedClassTable* table, DumpTimeSharedClassTable* clone) :
+                      _table(table), _cloned_table(clone) {
+    assert(_table != NULL, "_dumptime_table is NULL");
+    assert(_cloned_table != NULL, "_cloned_table is NULL");
+  }
+  bool do_entry(InstanceKlass* k, DumpTimeClassInfo& info) {
+    if (!info.is_excluded()) {
```

Comment:
Should assert_lock_strong(DumpTimeTable_lock) be added like at line 1574 below?

Reply:
should check lock, thank.